components/screens/full-pages/ledger/LedgerConnectForImport.tsx

```diff
@@ -51,77 +45,86 @@ export default function LedgerConnectForImport() {
     }
 
     if (transport && isAppOpen) {
-      setTimeout(() => {
-        navigate(redirect);
-      }, 1000); // Delay to prevent the page from flickering
-      return;
-    }
-
-    if (!callCheckNeedRetryOnce.current) {
-      callCheckNeedRetryOnce.current = true;
+      navigate(redirect);
       return;
     }
-
-    const timeout = setTimeout(() => {
-      if (!transport || !isAppOpen) {
-        setShowRetry(true);
-      }
-    }, 500); // Wait for the first attempt to connect
-    return () => clearTimeout(timeout);
   }, [transport, isAppOpen]);
 
+  const isLedgerConnected = !!transport || isPreviousConnected;
+
   return (
     <div className="flex h-[39rem] w-[41rem] flex-col items-stretch justify-between rounded-3xl bg-icy-blue-950 p-4 pb-6">
       <div className="space-y-4">
         <Header title="Connect Ledger" showClose={false} showPrevious={false} />
 
-        <div className="space-y-2">
-          <img
-            alt="ledger connecting"
-            className="mx-auto"
-            src={ledgerConnectingImage}
-          />
-          <div className="space-y-2 text-center">
-            <h1 className="text-xl font-semibold">
-              Please connect your Ledger device and open Kaspa app
-            </h1>
-            <p className="text-sm text-[#7b9aaa]"> Waiting for connection...</p>
+        <div className="space-y-10">
+          <div className="space-y-2">
+            <img
+              alt="ledger connecting"
+              className="mx-auto"
+              src={ledgerConnectingImage}
+            />
+            <div className="space-y-2 text-center">
+              <h1 className="text-xl font-semibold">
+                Please connect your Ledger device and open Kaspa app
+              </h1>
+            </div>
+          </div>
 
-            <div className="flex justify-center gap-4 text-center text-sm">
-              <div className="flex items-center gap-1">
-                <i
-                  className={twMerge(
-                    "hn text-sm",
-                    transport
-                      ? "hn-check-circle text-green-200"
-                      : "hn-exclamation-triangle text-red-400",
-                  )}
-                ></i>
-                Ledger connected
-              </div>
+          <div className="flex flex-col items-center justify-center gap-4 text-xs">
+            <div
+              className={twMerge(
+                "flex items-center gap-2 rounded-md p-2",
+                isLedgerConnected
+                  ? "bg-[#115E59] bg-opacity-30 text-[#14B8A6]"
+                  : "bg-[#122932]",
+              )}
+            >
+              {isLedgerConnected ? (
+                <i className="hn hn-check-circle text-xs text-[#14B8A6]"></i>
+              ) : (
+                <div
```

Comment:
@0x7u I used the spinner provided by preline; however, it is not working. I don't have idea to solve it. Can you take a look on it?

Reply:
Try this https://github.com/forbole/kastle/blob/bdd80c37d13165d67c52abaf22794e32f4416eb1/components/token-asset/TokenHistory.tsx#L41